servers/quarkus-server/src/main/java/org/projectnessie/server/config/VersionStoreConfig.java

```diff
@@ -52,4 +53,13 @@
     @WithName("db-path")
     String getDbPath();
   }
+
+  @ConfigMapping(prefix = "nessie.version.store.mongo")
+  interface MongoVersionStoreConfig {
+    @WithName("connect-to")
+    String getConnectionString();
```

Comment:
Maybe the `quarkus-mongodb-client` module already defines something that makes this superfluous.

Reply:
It does, but it will then configure its own instance of `MongoClient`. We currently have non-quarkus environments where we create our own `MongoClient`... Do you think it's worth refactoring the code to support both a locally created `MongoClient` and an injected one?

Reply:
Yea, it's probably worth to refactor the Quarkus-code to use the `MongoClient` produced by Quarkus.
Ran into the same issue for the Dynamo-adapter It felt really strange to have (a ton of) configuration options via Quarkus but only use a subset of those (authentication, endpoint-detection, aws-http-client config, etc). So I changed the Dynamo-PR that it uses the DynamoDbClient produced by Quarkus, but use the one "produced by us" in other places.

Reply:
refactored in the latest commit (also rebased)